test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -522,21 +522,23 @@ def obfuscate_snapshot(self, snap_dir, map_file, sudo_val):
                               "simply be deleted")
         jobspath = os.path.join(snap_dir, MOM_PRIV_PATH, "jobs")
         jbcontent = {}
-        for name in os.listdir(jobspath):
-            if name.endswith(".JB"):
-                ret = None
-                fpath = os.path.join(jobspath, name)
-                if printjob is not None:
-                    cmd = [printjob, fpath]
-                    ret = self.du.run_cmd(cmd=cmd, sudo=sudo_val,
-                                          as_script=True)
-                self.du.rm(path=fpath)
-                if ret is not None and ret["out"] is not None:
-                    jbcontent[name] = "\n".join(ret["out"])
-            # Also delete any other files/directories inside mom_priv/jobs
-            else:
-                path = os.path.join(jobspath, name)
-                self.du.rm(path=path, recursive=True, force=True)
+        jbfilelist = self.du.listdir(path=jobspath, sudo=sudo_val)
+        if jbfilelist is not None:
```

Comment:
Just a suggestion (because looks like you have used `is not None` in the whole file): `is not None` is not required here and other places you can simply do `if jbfilelist:`

Reply:
Why? I specifically want to test that it is not None .. "if X" will return True/False for a number of reasons .. probably doesn't make a difference here, but I think it's good practice to always check for exactly what you expect the function to return.

Reply:
In my view, self.du.listdir itself is wrong, in python, you should return only one return type not multiple from the method, and here self.du.listdir is returning either None or list, which it shouldn't. It should return list always and if the given path doesn't exist then it should either return an empty list or raise an exception. (Not sure but if AFAIK, this will be flagged as
But anyway, that's for future enhancement/refactoring of PTL, I am ok if you don't want to change right now, I was just giving suggestion, take it or leave it up to you 😄

Reply:
I agree, the functions should ideally return only 1 type. Thanks for the suggestions, they are very much welcome, but I'd like to keep this one as it is :)